src/wp-admin/includes/update-core.php

```diff
@@ -602,9 +617,13 @@ function update_core( $from, $to ) {
 		} else {
 			$lang_dir = WP_CONTENT_DIR . '/languages';
 		}
-
+		/*
+		 * Note: str_starts_with() is not used here, as this file is included
+		 * when updating from older WordPress versions, in which case
+		 * the polyfills from wp-includes/compat.php may not be available.
+		 */
 		// Check if the language directory exists first.
-		if ( ! @is_dir( $lang_dir ) && 0 === strpos( $lang_dir, ABSPATH ) ) {
+		if ( ! @is_dir( $lang_dir ) && str_starts_with( $lang_dir, ABSPATH ) ) {
```

Comment:
As per the docblock above I think this ma need reverting to `0 === strpos()`

Reply:
Reverted. Thanks.